src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -446,6 +446,14 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           case _ =>
         }
         checkClassType(tpt1.tpe, tpt1.pos, traitReq = false, stablePrefixReq = true)
+
+        tpt1 match {
+          case AppliedTypeTree(_, targs) =>
+            for (targ @ TypeBoundsTree(_, _) <- targs)
+              ctx.error("type argument must be fully defined", targ.pos)
```

Comment:
I think it would be better to not introduce new string error messages and to make a new error message class by following Felix's guide instead: https://github.com/lampepfl/dotty/issues/1589